### ISSUE TITLE
Currently no way to listen for changes to the URL

### ISSUE THREAD

**Issue by j-maas** [dillonkearns/elm-pages#358]
In a `Browser.application` I can listen to changes to the URL via `onUrlChange`. There is currently no equivalent for elm-pages.

This isn't urgent. See also the [discussion on Discord](https://discord.com/channels/534524278847045633/727578237101932556/1061685165476290600) for more details.

**Comment by jreusch4** [comment]
For reference, I also ran into that issue and did the following:

```ts
// index.ts
const config: ElmPagesInit = {
  async load(elmLoaded) {
    const app = await elmLoaded
    
    // override history methods that can modify the url
    for (const method of ['back', 'forward', 'go', 'pushState', 'replaceState']) {
      const original = history[method].bind(history)
      history[method] = (...args: any[]) => {
        original(...args)
        // send an update event to Elm
        app.ports.onUrlChange.send(location.pathname)
      }
    }
  }
}
```

**Comment by adamdicarlo** [comment]
Is this issue about elm-pages v3?

I've noticed, at least in v3, that `app/Shared.elm` defines a `template` function, and the `SharedTemplate` record includes `onPageChange : Maybe Msg`, so maybe this issue is fixed with elm-pages v3?

**Comment by jreusch4** [comment]
Yes, this does work in elm-pages v3!

It's been a few months, but I think I just did not find that option, so I stuck with my v2 solution instead. Thank you!

A small difference is that elm-pages sends this message after the new page is fully loaded, while overriding the history methods triggers the event as soon as the user clicked on a link. The elm-pages message also gets fired when the user presses the back button in the browser, while I did not do that in my Javascript solution (you would have to also listen for popstate/hashchange events)
